src/lib/comms/sol-socket-impl.h

```diff
@@ -52,8 +52,8 @@ struct sol_socket_impl {
     int (*bind)(struct sol_socket *s, const struct sol_network_link_addr *addr);
 };
 
-#ifdef SOL_PLATFORM_LINUX
-const struct sol_socket_impl *sol_socket_linux_get_impl(void);
+#ifdef SOL_PLATFORM_UNIX
+const struct sol_socket_impl *sol_socket_unix_get_impl(void);
```

Comment:
this is a good time to review this, @ibriano why did you use this instead of a `sol_socket_get_impl()` that is implemented per platform?